ext/standard/basic_functions.c

```diff
@@ -2285,17 +2259,27 @@ PHP_FUNCTION(getprotobynumber)
 /* {{{ Registers a tick callback function */
 PHP_FUNCTION(register_tick_function)
 {
-	user_tick_function_entry tick_fe;
+	user_tick_function_entry tick_fe = {
+		.fci_cache = empty_fcall_info_cache,
+		.params = NULL,
+		.param_count = 0,
+		.calling = false,
+	};
+	zend_fcall_info fci;
 	zval *params = NULL;
-	uint32_t param_count = 0;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "f*", &tick_fe.fci, &tick_fe.fci_cache, &params, &param_count) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "F*", &fci, &tick_fe.fci_cache, &params, &tick_fe.param_count) == FAILURE) {
```

Comment:
The change from `f` to `F` means that the function never worked for trampolines, right?

Reply:
It did because the callable was stored in the FCI, and it was fetched at each call from the `function_name` zval of the FCI.
I did think this as well at the beginning and so wrote the tests first.